nonlin_granger_casuality.py

```diff
@@ -0,0 +1,110 @@
+
+"""" https://pypi.org/project/nonlincausality/#description 
+     https://github.com/mrosol/Nonlincausality/blob/master/README.md
+"""
+import pandas as pd
+from nonlincausality import nonlincausalityNN as nlc_nn
+
+class NonlinearNNGrangerCausalityAnalysis:
+    """
+    This class performs nonlinear Granger causality analysis between pairs of stock tickers 
+    using the nonlincausalityNN method from the nonlincausality library. It applies neural 
+    networks to forecast and test for causality in time series data, allowing for the capture 
+    of nonlinear dependencies.
+    """
+    
+    def __init__(self, dataframe, max_lag=5, nn_config=['d','dr','d','dr'], nn_neurons=[100, 0.05, 100, 0.05]): #MLP Version
+        """
+        Initializes the NonlinearNNGrangerCausalityAnalysis object with stock data and configuration.
+
+        Args:
+            dataframe (pd.DataFrame): A DataFrame containing time series data of stock prices.
+            max_lag (int): The maximum number of lags to consider when testing for causality.
+            nn_config (list): List defining the configuration of the neural network layers.
+                - 'd' stands for dense layer, 'dr' stands for dropout regularization.
+            nn_neurons (list): List specifying the number of neurons in each layer and dropout rate.
+                - The format: [neurons, dropout_rate, neurons, dropout_rate, ...]
+        """
+        # Store the dataframe with stock data
+        self.dataframe = dataframe
+        # Extract the column names (tickers) from the DataFrame
```

Comment:
Make sure the dataframe we get has this, either check manually or add a check on the code (better add a check on the code)